design/gw-api-policy-and-filter-design.md

```diff
@@ -0,0 +1,126 @@
+The Contour Gateway Provisioner uses GatewayClass parameters for passing general configuration for an instance of Contour.
+
+Policy attachment is described in detail [here](https://gateway-api.sigs.k8s.io/references/policy-attachment/).
+Policy resources must follow a standard structure, referencing the resource they apply to via a [`targetRef`](https://gateway-api.sigs.k8s.io/references/policy-attachment/#target-reference-api) field.
+They could be "attached" to a Gateway, Route, Service (or other backend reference resource), or even a Namespace (howeer the resources a particular policy is allowed to attach to is implementation-dependent).
+A key feature of Policy attachment is the idea that Policies can be applied with a hierarchy.
+[Overrides and Defaults can be cascaded up/down the hierarchy of Gateway API resources](https://gateway-api.sigs.k8s.io/references/policy-attachment/#hierarchy).
+
+(Note that Policies could be attached to GatewayClasses, however it is [more complicated and probably not recommended](https://gateway-api.sigs.k8s.io/references/policy-attachment/#attaching-policy-to-gatewayclass))
+
+HTTPRoute also supports a per-route or per-backend [`HTTPRouteFilter`](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteFilter) type.
+This provides built-in HTTP request modification/redirection capabilities (varying between core and extended conformance levels).
+It also supports an `ExtensionRef` filter type, which can be to reference a custom resource for implementation-specific configuration.
+
+Policies and Filters represent extension points that could overlap in scope.
+Upstream Gateway API documentation provides some guidance [here](https://gateway-api.sigs.k8s.io/references/policy-attachment/#interaction-with-custom-route-filters).
+In particular the community recommends [differentiating what configurations are supported by each method](https://gateway-api.sigs.k8s.io/references/policy-attachment/#2-custom-filters-and-policies-should-not-overlap)
+
+## Goals
+- Differentiating which Contour features should be included in each configuration/extension type
+- Give maintainers/contributors/users an agreed upon framework/pattern in which to think about extension points in Gateway API so we can make the implementation and API consistent
+- Use abstractions/conventions of Gateway API as much as possible to avoid possible future of piecemeal attaching new features/configurations
+
+## Non Goals
+- Cover configuration of upstream TLS or other "backend protocol details"
+  - These will hopefully be covered by [this GEP](https://gateway-api.sigs.k8s.io/geps/gep-1282/) and subsequent features
+- Contour flags we've added for smaller surface area coverage or convention etc.
+  - e.g. `contourconfiguration.spec.envoy.listener.disableAllowChunkedLength`
+
+## High-Level Design
+Summary TBD
+
+## Detailed Design
+
+### Categorizing features between Policies or Filters
```

Comment:
Even just looking at the built-in filters -- e.g. the request header modifier -- in Contour/HTTPProxy, we support both global default header modifications, and per-route modifications. So to implement that in Gateway API, you'd need to use policy attachment, except there's also the built-in filter, and having both contradicts the guidance in https://gateway-api.sigs.k8s.io/references/policy-attachment/#2-custom-filters-and-policies-should-not-overlap. 🤔

Reply:
yep, that and what you mention with rate limiting + auth are what im struggling with a little, though there is a guideline about separating concerns, it doesnt seem like it is really that cut and dry with these examples

maybe something to bring up upstream